libs/libc/stdlib/lib_strtod.c

```diff
@@ -109,235 +181,462 @@ static bool chtod(char c, double base, FAR double *number)
     }
 
   *number = *number * base + tmp;
+
   return true;
 }
 
 /****************************************************************************
- * Public Functions
- ****************************************************************************/
-
-/****************************************************************************
- * Name: strtod
+ * Name: scalbnx
  *
  * Description:
- *   Convert a string to a double value
- *
- *   NOTE: This implementation is limited as compared to POSIX:
- *   - Hexadecimal input is not supported
- *   - INF, INFINITY, NAN, and NAN(...) are not supported
+ *   Implement functionality similar to scalbnl
```

Comment:
if so, why not call scalbnl directly

Reply:
because the avr-libc has no functions scalbnl and powl

Reply:
but should we use https://github.com/apache/nuttx/blob/master/libs/libc/math/lib_scalbnl.c?